apm-agent-plugins/apm-dubbo-plugin/src/test/java/co/elastic/apm/agent/dubbo/AbstractDubboInstrumentationTest.java

```diff
@@ -38,54 +39,82 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Callable;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.when;
 
 public abstract class AbstractDubboInstrumentationTest extends AbstractInstrumentationTest {
 
-    private final int port = TestPort.getAvailableRandomPort();
-    private final int anotherPort = TestPort.getAvailableRandomPort();
+    private static CoreConfiguration coreConfig;
 
     @Nullable
-    private DubboTestApi dubboTestApi;
+    private static DubboTestApi dubboTestApi;
 
-    static CoreConfiguration coreConfig;
+    private static int port = -1;
 
     @BeforeAll
     static void initInstrumentation() {
         coreConfig = tracer.getConfig(CoreConfiguration.class);
     }
 
     @BeforeEach
-    void startRootTransaction() {
-        when(coreConfig.getCaptureBody()).thenReturn(CoreConfiguration.EventType.OFF);
+    void beforeEach() {
+
+        if (null == dubboTestApi) {
+            // only start test dubbo once, but with delegation to subclass for creating it
+            // thus we can't do that in @BeforeAll
 
-        // using context classloader is required here
-        ClassLoader cl = Thread.currentThread().getContextClassLoader();
```

Comment:
In practice it does not seem to make any difference here, I don't know why (and if) it used to be required previously.